### PR TITLE
feat(test): Represent uncaught errors

### LINKED ISSUE #14449: Errors in tests still contain internal stack frames
```
error: Uncaught (in promise) AddrInUse: Address already in use (os error 48)
    const listener = Deno.listen({
                          ^
    at Object.opSync (deno:core/01_core.js:172:12)
    at opListen (deno:ext/net/01_net.js:45:17)
    at Object.listen (deno:ext/net/01_net.js:318:17)
    at Object.listen (deno:ext/net/04_net_unstable.js:12:18)
    at Server.listenAndServe (file:///Users/ib/dev/deno_std/http/server.ts:181:27)
    at serve (file:///Users/ib/dev/deno_std/http/server.ts:565:20)
    at file:///Users/ib/dev/deno_std/http/server_test.ts:680:24
    at testStepSanitizer (deno:runtime/js/40_testing.js:449:13)
    at asyncOpSanitizer (deno:runtime/js/40_testing.js:149:15)
    at resourceSanitizer (deno:runtime/js/40_testing.js:375:13)

```

Bottom three frames should have been filtered out. This was supposed to be fixed by https://github.com/denoland/deno/pull/14302 but seems that's not the case.

CC @nayeemrmn

### PR DESCRIPTION
**Pull request by nayeemrmn**
Closes #14449.
```ts
Deno.test("failing test", () => {
  throw new Error("message");
});

Deno.test("passing test", () => {});

Deno.test("test with dangling rejection", () => {
  Promise.reject(new Error("foo 3 message"));
});
```
image
cc @bartlomieju

### CODE REVIEW

File: cli/lsp/testing/execution.rs [hunk 1]
```diff
@@ -805,6 +809,10 @@ impl test::TestReporter for LspTestReporter {
     }
   }
 
+  fn report_uncaught_error(&mut self, _origin: &str, _error: &JsError) {
+    // TODO(nayeemrmn)
```
Comment (bartlomieju) [inline cli/lsp/testing/execution.rs:813]: @kitson any suggestions how we should handle this?

File: cli/tools/test.rs [hunk 2]
```diff
@@ -450,31 +479,63 @@ impl TestReporter for PrettyTestReporter {
...
+            "{}: {}",
+            colors::red_bold("error"),
+            format_test_error(js_error)
+          );
+          println!();
+          failure_titles.push(failure_title);
+        }
+        if let Some(js_error) = uncaught_error {
+          let failure_title =
+            format!("[{}]", self.to_relative_path_or_remote_url(&origin));
```
Comment (bartlomieju) [inline cli/tools/test.rs:526]: I suggest to remove these brackets and only use gray color so it looks similar to regular errors reports

**Review by bartlomieju — COMMENTED** [review]
Very nice!

File: cli/tools/test.rs (hunk 2 above)
Comment (nayeemrmn) [reply cli/tools/test.rs:526]: These are the titles for errors that most urgently need to be addressed, it's not appropriate to grey them. I considered even making them bold. How about this, I think it looks good:
```suggestion
            format!("{} (uncaught error)", self.to_relative_path_or_remote_url(&origin));
```

Comment (bartlomieju) [reply cli/tools/test.rs:526]: Sure, I think that's better

File: cli/lsp/testing/execution.rs [hunk 3]
```diff
@@ -805,6 +809,37 @@ impl test::TestReporter for LspTestReporter {
...
+        TestOrTestStepDescription::TestStepDescription(desc) => {
+          self.progress(lsp_custom::TestRunProgressMessage::Failed {
+            test: desc.into(),
+            messages: messages.clone(),
+            duration: None,
+          });
+        }
+      }
+    }
+  }
```
Comment (nayeemrmn) [inline cli/lsp/testing/execution.rs:841]: I did the lsp part by just making it fail any running test from that origin. cc @kitsonk

**Comment by bartlomieju** [comment]
@nayeemrmn I tested this PR in VSCode, but I don't see a difference between this build and latest Deno, all I get is a prompt like this:
Screenshot 2022-05-07 at 16 47 13

**Comment by nayeemrmn** [comment]
> @nayeemrmn I tested this PR in VSCode, but I don't see a difference between this build and latest Deno, all I get is a prompt like this:
> Screenshot 2022-05-07 at 16 47 13

I had forgotten to update the worker orchestration on the lsp side, it's done now.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**